tests/docker/test_login.py

```diff
@@ -0,0 +1,41 @@
+"""Test login against custom registries."""
+
+from pathlib import Path
+from unittest.mock import MagicMock, call
+
+import supervisor.docker
+
+
+def test_login_configfile_exists(docker):
+    """Test if the docker.login gets called with the arguments matching the config file."""
+    old_dockercfg = supervisor.docker.DOCKERCFG
+
+    supervisor.docker.DOCKERCFG = (
+        Path(__file__).parents[1].joinpath("fixtures/dockercfg.json")
+    )
+    docker.docker.login = MagicMock(name="login")
+
+    docker._login()
+
+    calls = [
+        call(registry="home-assistant.io", username="user_1", password="pass_1"),
+        call(registry="example.com", username="user_2", password="pass_2"),
+    ]
+
+    docker.docker.login.assert_has_calls(calls, any_order=True)
+
+    supervisor.docker.DOCKERCFG = old_dockercfg
+
+
+def test_login_configfile_folder(docker):
```

Comment:
The reason behind this test is that if the mounted file doesn't exist on the host, docker creates a folder in the target container with that name.